src/mlpack/tests/reward_clipping_test.cpp

```diff
@@ -63,13 +64,7 @@ BOOST_AUTO_TEST_CASE(RewardClippedAcrobotWithDQN)
   for (size_t trial = 0; trial < 3; ++trial)
   {
     // Set up the network.
-    FFN<MeanSquaredError<>, GaussianInitialization> model(MeanSquaredError<>(),
-        GaussianInitialization(0, 0.001));
-    model.Add<Linear<>>(4, 64);
-    model.Add<ReLULayer<>>();
-    model.Add<Linear<>>(64, 32);
-    model.Add<ReLULayer<>>();
-    model.Add<Linear<>>(32, 3);
+    VanillaDQN<> model(4, 64, 32, 3);
```

Comment:
Do you think we should show in a single test, it's also possible to manually specifiy the network, completely without using `VanillaDQN`?

Reply:
Hmm. I think u mean this:
```
    FFN<MeanSquaredError<>, GaussianInitialization>
        network(MeanSquaredError<>(),
                GaussianInitialization(0, 0.001));
    network.Add<Linear<>>(6, 256);
    network.Add<ReLULayer<>>();
    network.Add<Linear<>>(256, 3);

    // Create custom network type
    VanillaDQN<decltype(network)> model(std::move(network));
```
This is taken from the test for `DoublePoleCartWithDQN`, showing how we can manually specify the network. But I could add a separate test for CartPole with DQN as well, showing how to manually specify.?

I'm afraid we wont be able to directly pass the network to `QLearning`(ie without using `VanillaDQN`), as I've used the method `ResetParametersIfEmpty()` in QLearning which is not present in FNN. Would that be fine?

Reply:
We could rename `ResetParametersIfEmpty` to `ResetParameters` and revert the change, that way we can still pass a vanilla network, maybe I missed something? There is already a test that uses the copy constructor, but would be great if we could provide backward compatibility.

Reply:
Yeah, you are right.. Actually I didn't notice that `Parameters()` had already been added to `SimpleDQN`.. 
Anyways, now I've done the necessary changes.. Kindly have a look..